openapi/index.yaml

```diff
@@ -172,79 +172,120 @@ definitions:
       expiration_date:
         $ref: "#/definitions/Timestamp"
     required:
-      - status
       - activation_date
       - expiration_date
-  CgnRevokedStatus:
-    type: object
-    properties:
-      status:
-        type: string
-        x-extensible-enum:
-          - REVOKED
-      activation_date:
-        $ref: "#/definitions/Timestamp"
-      expiration_date:
-        $ref: "#/definitions/Timestamp"
-      revocation_date:
-        $ref: "#/definitions/Timestamp"
-      revocation_reason:
-        type: string
-        minLength: 1
-    required:
-      - status
-      - revocation_date
-      - revocation_reason
-      - activation_date
-      - expiration_date
-  CgnExpiredStatus:
+    
+  CardPendingStatus:
     type: object
     properties:
       status:
         type: string
         x-extensible-enum:
-          - EXPIRED
-      activation_date:
-        $ref: "#/definitions/Timestamp"
-      expiration_date:
-        $ref: "#/definitions/Timestamp"
+          - PENDING
     required:
       - status
-      - activation_date
-      - expiration_date
-  CgnStatus:
+  CardActivatedStatus:
+    allOf:
+      - $ref: "#/definitions/CommonCardStatus"
+      - type: object
+        properties:
+          status:
+            type: string
+            x-extensible-enum:
+              - ACTIVATED
+        required:
+          - status
+  CardRevokedStatus:
+    allOf:
+      - $ref: "#/definitions/CommonCardStatus"
+      - type: object
+        properties:
+          status:
+            type: string
+            x-extensible-enum:
+              - REVOKED
+          revocation_date:
+            $ref: "#/definitions/Timestamp"
+          revocation_reason:
+            type: string
+            minLength: 1
+        required:
+          - status
+          - revocation_date
+          - revocation_reason
+  CardExpiredStatus:
+    allOf:
+      - $ref: "#/definitions/CommonCardStatus"
+      - type: object
+        properties:
+          status:
+            type: string
+            x-extensible-enum:
+              - EXPIRED
+        required:
+          - status
+  CardStatus:
     x-one-of: true
     allOf:
-      - $ref: "#/definitions/CgnPendingStatus"
-      - $ref: "#/definitions/CgnActivatedStatus"
-      - $ref: "#/definitions/CgnRevokedStatus"
-      - $ref: "#/definitions/CgnExpiredStatus"
+      - $ref: "#/definitions/CardPendingStatus"
+      - $ref: "#/definitions/CardActivatedStatus"
+      - $ref: "#/definitions/CardRevokedStatus"
+      - $ref: "#/definitions/CardExpiredStatus"
     x-example: 
       status: ACTIVATED
       activation_date: "2021-10-13T00:00:00.000Z"
       expiration_date: "2030-10-13T00:00:00.000Z"
-    
+
+  EycaCardActivatedStatus:
+    allOf:
+      - $ref: "#/definitions/CardActivatedStatus"
+      - type: object
+        properties:
+          card_number:
+            type: string
+            minLength: 1
+        required:
+          - card_number
```

Comment:
Maybe we should rename this in `EycaCardActivated` or `ActivatedEycaCard`, meaning this is an EycaCard with activated status. The `-Status` prefix let me think it's just the status value

Reply:
Yep, the rename impacts several files :D